docs/architecture/feature-modules.md

```diff
@@ -0,0 +1,522 @@
+- **Message List Glance**: Glanceable message widget
+  - **API**: Public interfaces for glance widget
+  - **Implementation**: Concrete implementations of glance widget
+- **Shortcut**: App shortcut widgets
+  - **API**: Public interfaces for shortcut widgets
+  - **Implementation**: Concrete implementations of shortcut widgets
+- **Unread**: Unread message counter widget
+  - **API**: Public interfaces for unread counter widget
+  - **Implementation**: Concrete implementations of unread counter widget
+
+## 🔄 Supporting Feature Modules
+
+In addition to the core email functionality, the application includes several supporting feature modules:
+
+### 🔎 Autodiscovery Module
+
+The Autodiscovery module automatically detects email server settings.
+
+#### Subfeatures:
+
+- **API** (`feature:autodiscovery:api`): Public interfaces
+- **Autoconfig** (`feature:autodiscovery:autoconfig`): Automatic configuration
+- **Service** (`feature:autodiscovery:service`): Service implementation
+- **Demo** (`feature:autodiscovery:demo`): Demonstration implementation
+
+### 💰 Funding Module
+
+The Funding module handles in-app donations and funding options.
+
+#### Subfeatures:
+
+- **API** (`feature:funding:api`): Public interfaces
+- **Google Play** (`feature:funding:googleplay`): Google Play billing integration
+- **Link** (`feature:funding:link`): External funding link handling
+- **Noop** (`feature:funding:noop`): No-operation implementation
+
+### 🔄 Migration Module
+
+The Migration module handles data migration between different email clients.
+
+#### Subfeatures:
+
+- **Provider** (`feature:migration:provider`): Migration data providers
+- **QR Code** (`feature:migration:qrcode`): QR code-based migration
+- **Launcher** (`feature:migration:launcher`): Migration launcher
+  - **API** (`feature:migration:launcher:api`): Launcher interfaces
+  - **Noop** (`feature:migration:launcher:noop`): No-operation implementation
+  - **Thunderbird** (`feature:migration:launcher:thunderbird`): Thunderbird-specific implementation
+
+### 📊 Telemetry Module
+
+The Telemetry module handles usage analytics and reporting.
+
+#### Subfeatures:
+
+- **API** (`feature:telemetry:api`): Public interfaces
+- **Noop** (`feature:telemetry:noop`): No-operation implementation
+- **Glean** (`feature:telemetry:glean`): Mozilla Glean integration
+
+## 🔌 Extending with Additional Features
+
+The modular architecture of Thunderbird for Android allows for easy extension with additional features. Here are some examples of how the app could be extended:
+
+### 📅 Calendar Feature
+
+A Calendar feature could be added to integrate calendar functionality with email.
+
+```
+feature:calendar
+├── feature:calendar:api
+├── feature:calendar:impl
+├── feature:calendar:event
+│   ├── feature:calendar:event:api
+│   └── feature:calendar:event:impl
+└── feature:calendar:sync
+    ├── feature:calendar:sync:api
+    └── feature:calendar:sync:impl
+```
+
+### ✅ Todo Feature
```

Comment:
I think we should condense this to 1-2 examples. They are all very similar, so more than that doesn't make it more clear.